### DISCUSSION TITLE
Screen Locker

### DISCUSSION THREAD

**Discussion by ackstorm23** [hyprwm/Hyprland#179] [Q&A]
What are those of you using Hyprland using to lock your screen?

**Comment by vaxerski** [comment]
swaylock, i use a fork with some added effects [https://github.com/mortie/swaylock-effects](https://github.com/mortie/swaylock-effects)

**Reply by rafay99-epic** [reply]
It's awesome, I use is using the terminal.

**Comment by ackstorm23** [comment]
Do you just manually lock then, or how do you trigger automatic lock after an idle period?

**Reply by vaxerski** [reply]
manually, you can use swayidle for automatic locks.

**Reply by ackstorm23** [reply]
I have been trying to use swayidle as I used in sway, but it immediate goes defunct if it run it as exec= or exec-once= in hyprland

**Reply by vaxerski** [reply]
launch it in a late launch

**Reply by vollowx** [reply]
> launch it in a late launch

how can I do that

**Reply by zcorn2017** [reply]
> > launch it in a late launch
> 
> how can I do that

a workaround found.

```

exec-once = swayidle & disown

```

`&` means detaching the process.
And use `disown` to disown the process to free it from your terminal.
That's it.
Half a day wasted on such a silly thing...

**Reply by vaxerski** [reply]
odd

**Reply by vollowx** [reply]
> > > launch it in a late launch
> > 
> > 
> > how can I do that
> 
> a workaround found.
> 
> ```
> 
> exec-once = swayidle & disown
> ```
> 
> `&` means detaching the process. And use `disown` to disown the process to free it from your terminal. That's it. Half a day wasted on such a silly thing...

![image](https://user-images.githubusercontent.com/73375859/219676506-edf5083a-8b4c-4022-b32e-b66694fb77db.png)

It runs normally, so maybe I do not need ` & disown`?
